### DISCUSSION TITLE
Dapat menerima pesan dalam bentuk Encoded URL

### DISCUSSION THREAD

**Discussion by doelmi** [kirimwa-id/forum#72] [Saran dan Ide]
Mohon pesan yang berbentuk encoded URL dapat diterjemahkan secara langsung seperti yang ada di api WA (https://api.whatsapp.com/).

Contoh pesan saya yang berbentuk encoded URL adalah
`_Assalamu%27alaikum%20Warahmatullahi%20Wabarakaatuh_%0A%0ATanpa%20mengurangi%20rasa%20hormat%2C%20izinkan%20kami%20mengundang%20%2A%7Bnickname%7D%2A%20untuk%20hadir%20serta%20memberikan%20do%27a%20restu%20pada%20acara%20pernikahan%20kami.%0A%0AUntuk%20detail%20acara%2C%20lokasi%2C%20dan%20ucapan%20bisa%20klik%20link%20dibawah%20ini%3A%0A%7Blink_code%7D%0A%0AMerupakan%20suatu%20kehormatan%20dan%20kebahagiaan%20bagi%20kami%2C%20apabila%20%2A%7Bnickname%7D%2A%20berkenan%20hadir.%0A%0ADo%27a%20restu%20Anda%20merupakan%20hadiah%20terindah%20bagi%20kami.%0A%0AAtas%20kehadiran%20dan%20do%27a%20restu%20yang%20telah%20diberikan%2C%20kami%20ucapkan%20terima%20kasih.%0A%0A_Wassalamu%27alaikum%20Warahmatullahi%20Wabarakatuh._`

sehingga apabila dikirimkan akan menjadi
![image](https://user-images.githubusercontent.com/26380952/144538853-35ecc55c-9fa7-4c93-b64e-9d1263ba9106.png)

Isi json saya:
![image](https://user-images.githubusercontent.com/26380952/144539202-08bf6826-d3fb-44a7-a42c-998687de0691.png)

Saya tahu kalau bisa menggunakan \n untuk enter, tapi terasa lebih ribet karena tidak ada fungsi yang mengubah enter menjadi \n di PHP.

**Yang diharapkan**
Parameter text query yang ada di api WA dapat menerjemahkan pesan yang ter-encode sehingga ketika ada enter dll, bisa di-translate juga dan menjadi hasil yang diinginkan.

![image](https://user-images.githubusercontent.com/26380952/144539114-e8dfc5b1-9480-4dc7-8650-d9f9a1b3cab7.png)

![image](https://user-images.githubusercontent.com/26380952/144538975-4d5b4653-8e0d-4d94-9ff3-57bd8e2589d9.png)

**Comment by rioastamal** [comment]
@doelmi,

Pesan yang dikirimkan ke API KirimWA.id adalah murni tanggung jawab developer. Misal kode yang dikirim masih berbentuk encoded string maka tugas developer adalah melakukan decode pesan tersebut sebelum dikirimkan ke API KirimWA.id.

Misal jika anda pengguna PHP maka pesan tersebut dapat di-decode terlebih dulu dengan fungsi `urldecode()`.

```php
 $decodedString = urldecode("_Assalamu%27alaikum%20Warahmatullahi%20Wabarakaatuh_%0A%0ATanpa%20mengurangi%20rasa%20hormat%2C%20izinkan%20kami%20mengundang%20%2A%7Bnickname%7D%2A%20untuk%20hadir%20serta%20memberikan%20do%27a%20restu%20pada%20acara%20pernikahan%20kami.%0A%0AUntuk%20detail%20acara%2C%20lokasi%2C%20dan%20ucapan%20bisa%20klik%20link%20dibawah%20ini%3A%0A%7Blink_code%7D%0A%0AMerupakan%20suatu%20kehormatan%20dan%20kebahagiaan%20bagi%20kami%2C%20apabila%20%2A%7Bnickname%7D%2A%20berkenan%20hadir.%0A%0ADo%27a%20restu%20Anda%20merupakan%20hadiah%20terindah%20bagi%20kami.%0A%0AAtas%20kehadiran%20dan%20do%27a%20restu%20yang%20telah%20diberikan%2C%20kami%20ucapkan%20terima%20kasih.%0A%0A_Wassalamu%27alaikum%20Warahmatullahi%20Wabarakatuh._");
```
